### ISSUE TITLE
obstacle_time values are silently ignored for some tags

### ISSUE THREAD

**Issue by Zero3** [osmandapp/OsmAnd#12854]
### Description

`obstacle_time` values in `routing.xml` are silently ignored for some kind of tags. This leads to wrongly estimated travel times.

I can reproduce this for the `amenity` and `name` tags, but the issue probably happens for other tags as well.

I suspect this happens because OsmAnd has some kind of special treatment of these tags, perhaps because it uses them for other purposes as well, and then "forgets" them when processing stuff like `obstacle_time` during routing.

### How to reproduce?

Add the following test statement to the top of the `obstacle_time` section in `routing.xml`:

````
<select value="100000" t="amenity" v="ferry_terminal"/>
````

Then test a route that goes through a node with the `amenity=ferry_terminal` tag. For example this one:

Origin: https://www.openstreetmap.org/node/111775427
Destination: https://www.openstreetmap.org/node/3579875494

Observe that the ~28 hour penalty is not added to the travel time estimate.

If the `obstacle_time` statement is modified to target a tag not affected by this bug, one will see that the ~28 hour penalty is added to the travel time as intended:

````
<select value="100000" t="ferry" v="yes"/>
````

### Your Environment
OsmAnd Version: 4.0.8

**Comment by xmd5a2** [comment]
They are ignored because they are not in the obf data. You should update rendering_types.xml `<category name="routing">` and rebuild maps. Or you can create a pull request and we will consider it.

**Comment by Zero3** [comment]
@xmd5a2 I see. I am not familiar with that part of OsmAnd (I was not even aware of anything called 'obf'), so perhaps someone else can make a pull request for the necessary changes.

In my case, I want to be able to assign penalties to `amenity=ferry_terminal` in my `routing.xml`, since taking a ferry entails a significant amount of time wasted on waiting, ticketing, boarding, unboarding, ....

**Comment by Zero3** [comment]
@xmd5a2 Is this supposed to be fixed in version 4.1.10 of the Android app? Because I tested this issue again on version 4.1.10, and it still does not work. No time is added to the estimated travel time for a route passing through multiple `amenity=ferry_terminal` nodes, even with a huge penalty like this one:

````
<osmand_routing_config ...>
	<routingProfile ...>
		...
		<point attribute="obstacle_time">
			...
			<select value="1800" t="amenity" v="ferry_terminal"/>
			...
		</point>
		...
	</routingProfile>
</osmand_routing_config>
````

You mention something about "rebuilding maps". Is this something that every single `OsmAnd` user has to do manually for the change to work?

**Comment by vshcherb** [comment]
Maps already have that information, so no need to rebuild the maps

**Comment by Zero3** [comment]
@vshcherb could you please reopen this issue? It is still not working, per my comment above. Thanks.

**Comment by vshcherb** [comment]
The issue is already released with 4.1, so unfortunately we don't plan any 4.1 releases any more

**Comment by Zero3** [comment]
@vshcherb I am a bit confused. How do you want to track this bug then? Should I create a new Github issue for the bug for it to get fixed?

**Comment by vshcherb** [comment]
Yes, you need to create new issue and provided updated information if possible, as it mentioned here ferry_terminal should be already included - https://github.com/osmandapp/OsmAnd-resources/commit/c259c16c14774813623a1e3c1fec8138c9b9f856. So probably the issue is somehow different than expected
